scrapy/utils/misc.py

```diff
@@ -184,6 +184,33 @@ def set_environ(**kwargs):
                 os.environ[k] = v
 
 
+class GeneratorNodeVisitor(ast.NodeVisitor):
+    """Visits all AST nodes of a generator, but ignores functions defined inside the generator.
+    If the generator code includes a return statement then sets includes_return_statement to True.
+    """
+    def __init__(self):
+        self.read_func_header = False
+        self.includes_return_statement = False
+        self.return_statement_lineno = None
```

Comment:
How is `return_statement_lineno` used?

Reply:
Forgot to remove it, sorry. I was gonna add the line number to the error message, but then decided to limit the change.
I remove it now.